### PR TITLE
Move to v2 module and enable stream redirection

### PR DESCRIPTION
**Pull request by alexellis**
## Description

Move to v2 module and enable stream redirection

* Moves to v2 module style as per google/github-go
* Allows stdio buffer to be disabled for large outputs
* Allows stdio reader/writer replacement fixes: #12
* Closes: #13 

## How Has This Been Tested?

```golang
package main

import (
	"bytes"
	"context"
	"fmt"

	gexec "github.com/alexellis/go-execute/v2"
)

func main() {
	buf := bytes.NewBuffer(nil)

	t := gexec.ExecTask{
		Command:            "whoami",
		Shell:              true,
		StreamStdio:        false,
		StdOutWriter:       buf,
		DisableStdioBuffer: true,
	}

	r, err := t.Execute(context.Background())
	if err != nil {
		panic(err)
	}

	if r.ExitCode != 0 {
		panic(r.Stderr)
	}

	fmt.Printf("r.Stdout: %q\n", r.Stdout)
	fmt.Printf("Buffer: %q\n", buf.String())
}

```

Output:

```
go run .
r.Stdout: "alex\n"
Buffer: "alex\n"
```

Also tested with shell enabled/disabled, and with StreamStdio enabled.

## Checklist:

I have:

- [x] updated the documentation and/or roadmap (if required)

### CODE REVIEW

**Comment by alexellis** [comment]
@josegonzalez

**Comment by alexellis** [comment]
@michaeladler @josegonzalez

File: exec.go [hunk 1]
```diff
@@ -72,7 +87,18 @@ func (et ExecTask) Execute(ctx context.Context) (ExecResult, error) {
 	var command string
 	var commandArgs []string
 	if et.Shell {
-		command = "/bin/bash"
+
+		// On a NixOS system, /bin/bash doesn't exist at /bin/bash
```
Comment (michaeladler) [inline exec.go:91]: Correct, although the same is true for the *BSD systems, e.g. on OpenBSD bash (if installed) resolves to /usr/local/bin/bash.

**Comment by michaeladler** [comment]
LGTM

File: exec.go (hunk 1 above)
Comment (alexellis) [reply exec.go:91]: TIL

**Pull request merged**